README.md

```diff
@@ -1247,6 +1250,140 @@ echo 'Employee name: '.$employee->getName(); // Employee name: John Doe
 
 **[⬆ back to top](#table-of-contents)**
 
+## Error handling
+
+### Use Exceptions Rather Than Return Codes
+
+**Bad:**
+
+```php
+class DeviceController 
+{
+    const DEV1 = "/dev/dev1";
+    private $record;
+
+    public function sendShutDown(): void
+    {
+        $handle = $this->getHandle(self::DEV1);
+
+        if ($handle != DeviceHandle::INVALID) {
+            $this->record = $this->retriveDeviceRecord($handle);
+
+            if ($this->record->getStatus() != self::DEVICE_SUSPENDED) {
+                $this->pauseDevice($handle);
+                $this->clearDeviceWorkQueu();
+            } else {
+                Log::debug("Device suspended. Unable to shut down");
+            }
+        } else {
+            Log::debug("Invalid handle for: ". self::DEV1);
+        }
+    }
+
+    private function getHandle($deviceId): DeviceHandle
+    {
+        return new DeviceHandle($devicdId);
+    }
+
+    private function retriveDeviceRecord(DeviceHandle $handle): DeviceRecord
+    {
+        return new DeviceRecord($handle);
+    }
+}
```

Comment:
There is too much going on in this example. Shutting down, factories, logging, record validation.
I can't even find the *return code* part.

Example with as less code as possible expressing only the point you try to explain would be better.